csrc/host_ir/pass/insert_deallocations.cpp

```diff
@@ -33,6 +33,11 @@ void InsertDeallocations::passImplementation(Fusion* fusion) {
     }
   }
 
+  // Remove outputs from last_use, they should not be deallocated
+  for (auto* out : ir_utils::filterByType<TensorView>(hic->outputs())) {
+    last_use.erase(out);
+  }
+
   std::vector<std::pair<int64_t, TensorView*>> last_use_by_index;
```

Comment:
```suggestion
  for (auto* out_val : hic->outputs()) {
    if (!out_val->isA<TensorView>()) {
      continue;
    }
    auto* out = out_val->as<TensorView>();
```
**[Suggested by AI]** The loop logic was rewritten to avoid using the ­`ir_utils::filterByType<TensorView>` helper.

Old code
• Relied on ­`filterByType<TensorView>` to return only `TensorView*` objects.

New code
• Iterates over every value returned by `hic->outputs()`.
• Explicitly skips non-TensorView values with an `isA<TensorView>()` check.
• Casts the surviving values to `TensorView*` with `as<TensorView>()`.

Effect
• Removes the dependency on `filterByType`, eliminating the compile-time error and making the type filtering explicit and local to the loop.

Reply:
bot has no understanding about the logic, all it tried was to fix the build 😄